lib/background_worker/base.rb

```diff
@@ -72,127 +66,26 @@ def perform_in_background(method_name, options={})
-        Resque.redis.client.disconnect
-        Resque.redis.client.reconnect
-        Rails.cache.reconnect
+        options[:uid]
       end
 
-
       # This method is called by the job runner
       #
       # It will just call your preferred method in the worker.
-      def perform(method_name, options={})
-        rescue_reconnect if defined?(Resque)
-        ActiveRecord::Base.verify_active_connections!
-
-        raise ArgumentError, ":uid is required: Options given were: #{options.inspect}" if options['uid'].blank?
-        setup_logger(method_name)
-        raise ArgumentError, ":current_user_id is required: Options given were: #{options.inspect}" if options['current_user_id'].blank?
-        set_current_user(options['current_user_id'])
-
-        worker = self.new(options)
-        worker.report_progress "Task started"
-
-        returned_data = worker.send(method_name, options)
-
-        worker.log "returned data : #{returned_data.inspect}"
-        worker.log "completed : #{worker.state.completed}"
-        if !worker.state.completed
-          worker.log "data : #{worker.state.data}"
-          worker.state.data.merge!(hasherize_opts(returned_data, :key_if_not_hash => :result))
-          worker.log "data : #{worker.state.data}"
-          worker.report_successful
-          worker.log "report_sucessful"
-        end
-
-      rescue Exception => e
-        puts "ERROR: #{e} \n #{e.backtrace.join("\n")}"
-        if logger
-          log("#{options['uid']}: Exception: #{e}", :error)
-          log("#{options['uid']}: Invalid Record: #{e.record.errors.full_messages.to_sentence}", :error) if e.is_a?(ActiveRecord::RecordInvalid)
-          log("#{options['uid']}: #{e.backtrace.join("\n")}", :error)
-        end
-
-        BackgroundWorker.after_exception.call(e) if BackgroundWorker.after_exception
-
-        if worker && !worker.state.completed
-          worker.report_failed "An unhandled exception occurred: #{e}"
-        end
-      end
-
-      # LEGACY METHOD -- use instead of Resque.enqueu if you wanna spawn later...
-      # Fork off, and do some work!
-      #
-      # Tell a spawning worker class to instantiate itself, and run a method
-      def forked_send(method_name, options={})
-        worker_name = "#QT-WRK# #{self}::#{method_name}"
-         
-        # Fork off...
-        #
-        # Note: exceptions are trapped in here.
-        worker.spawn(:kill => true, :method => :fork, :argv => worker_name, :nice => 5) do
-          worker.perform(method_name, options)
-        end
-      end
-
+      def perform(method_name, options = {})
+        BackgroundWorker.verify_active_connections!
 
-      # Setup logger
-      #  - is ok as a class variable, as we should be forked out
-      def setup_logger(method)
-        log_dir = "#{Rails.root.to_s}/log/background_jobs"
-        Dir.mkdir(log_dir) unless File.exist?(log_dir)
+        # Special 'user' handling
+        store_current_user_in_thread(options['current_user_id'])
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.